software/control/microcontroller.py

```diff
@@ -630,21 +639,20 @@ def read_received_packet(self):
             self._cmd_id_mcu = msg[0]
             self._cmd_execution_status = msg[1]
             if (self._cmd_id_mcu == self._cmd_id) and (self._cmd_execution_status == CMD_EXECUTION_STATUS.COMPLETED_WITHOUT_ERRORS):
-                if self.mcu_cmd_execution_in_progress == True:
+                if self.mcu_cmd_execution_in_progress:
                     self.mcu_cmd_execution_in_progress = False
                     self.log.debug("mcu command " + str(self._cmd_id) + " complete")
-            elif self._cmd_id_mcu != self._cmd_id and time.time() - self.last_command_timestamp > 5 and self.last_command != None:
```

Comment:
The way this was implemented before, once we timed out we'd check the next 9 messages and finally exit if we didn't get the command ack.  This is a bit odd, so instead we turned the timeout way down and combine the checksum + ack timeout together into a combined retry count.

If we get >MAX_RETRY_COUNT of the combination of the two, we abort the command.